### ISSUE TITLE
Inherited method returning tuple with TypeVar is incorrectly marked as error

### ISSUE THREAD

**Issue by hjalmarlucius** [microsoft/pyright#2430]
**Describe the bug**
Method inheritance reports errors for correct TypeVar usage when wrapped as tuple.

**Screenshots or Code**
If applicable, add screenshots or the text of the code (surrounded by triple back ticks) to help explain your problem.
```python
from typing import TypeVar

T = TypeVar("T")

class A:
    def f(self, x: T, y: float) -> tuple[T, float]:
        ...

class B(A):
    def f(self, x: T, y: float) -> tuple[T, float]:
        ...
```

Error:
```
error: Method "f" overrides class "A" in an incompatible manner
    Return type mismatch: base method returns type "tuple[T@f, float]", override returns type "tuple[T@f, float]"
      Tuple entry 1 is incorrect type
        Type "T@f" cannot be assigned to type "T@f" (reportIncompatibleMethodOverride)
```

**Comment by erictraut** [comment]
I'm not able to repro the problem with your sample code. Can you please confirm that you're running the latest version of pyright (1.1.178)?

**Comment by hjalmarlucius** [comment]
Apologies, I was running 1.1.155 - no problems with the latest version
